src/endpoint/s3/s3_rest.js

```diff
@@ -190,7 +190,7 @@ async function _get_redirection_bucket(req, bucket) {
     if (redirect) {
         const dest = redirect.host_name;
         const protocol = redirect.protocol || req.secure ? 'https' : 'http';
```

Comment:
Better add parenthesis to separate between two possible operator precedence options:
```
        const protocol = (redirect.protocol || req.secure) ? 'https' : 'http';
```
vs.
```
        const protocol = redirect.protocol || (req.secure ? 'https' : 'http');
```